src/models/contact/index.js

```diff
@@ -87,3 +90,44 @@ export async function addContactIndexToAddressBook(addressBook, type, fetch) {
     thing: getThing(dataset, asUrl(addressBook.thing, datasetUrl)),
   };
 }
+
+export async function saveContact(addressBook, contactSchema, type, fetch) {
+  const { respond, error } = createResponder();
+  const { containerUrl: addressBookContainerUrl } = addressBook;
+  const indexIri = getContactIndexUrl(addressBook, type);
+  const newContact = await type.createContact(
+    addressBookContainerUrl,
+    contactSchema
+  );
+  const { iri } = newContact;
+  const { response: contact, error: saveContactError } = await saveResource(
+    newContact,
+    fetch
+  );
+  if (saveContactError) return error(saveContactError);
+
+  const contactIndex = await getContactIndexDataset(addressBook, type, fetch);
+
+  const contactThing = defineThing(
+    {
+      url: `${getSourceUrl(contact)}#this`,
+    },
+    (t) =>
+      addStringNoLocale(t, vcard.fn, contactSchema.fn || contactSchema.name),
+    (t) => addUrl(t, vcardExtras("inAddressBook"), indexIri)
+  );
+
+  const contactIndexIri = getSourceUrl(contactIndex);
+  const contactResource = {
+    dataset: setThing(contactIndex, contactThing),
+    iri: contactIndexIri,
+  };
+
+  const { response: contacts, error: saveContactsError } = await saveResource(
+    contactResource,
+    fetch
+  );
+
+  if (saveContactsError) return error(saveContactsError);
+  return respond({ iri, contact, contacts });
```

Comment:
I don't want us to use the `{ response, error }` pattern in the model modules - It should return the standardized object or objects of whatever the action has affected.

In this case saving a contact affects the new resource and thing that represents the newly created contact. We also need to ensure that the addressbook exists, and that the people index is exists, as these are created on the fly when needed. That means, as this code stands, it will work when you already have an address book and people index created, but I suspect it won't when that's the case.

Check out `saveGroup` in [src/models/contact/group/index.js](https://github.com/inrupt/pod-browser/blob/main/src/models/contact/group/index.js#L80-L120), that's a good example that shows you what might happen. The corresponding tests also shows what you need to check, namely that it works for a newly created address book that isn't saved in a resource (yet), i.e. an uninstantiated address book, and one that is. You can also see from the that method that it returns a complex object `{ addressBook, group, groupIndex }`, and I suspect we have to do this here as well (but then `{ addressBook, person, personIndex }`).